### PR TITLE
2_issues_fixed

### LINKED ISSUE #5732: Example of EIP-1559 Transfer for ERC20 token
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What would you like to say about Web3.js' documentation?

This particular example of an EIP-1559 transaction to transfer ER20 token from one address to another, which should be very common (e.g. for sending USDC), is not covered in the docs and I googled the entire web with no results. Can you please help?

### Examples/References

_No response_

### PR DESCRIPTION
**Pull request by sarthak1dev**
## Description
This pr fixes: #5732 and #6866, as requested by @jdevcs in the previous pr

Please include a summary of the changes and be sure to follow our [Contribution Guidelines](https://github.com/web3/web3.js/blob/4.x/.github/CONTRIBUTING.md).

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] I have selected the correct base branch.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have made corresponding changes to the documentation.
- [ ] My changes generate no new warnings.
- [ ] Any dependent changes have been merged and published in downstream modules.
- [ ] I ran `npm run lint` with success and extended the tests and types if necessary.
- [ ] I ran `npm run test:unit` with success.
- [ ] I ran `npm run test:coverage` and my test cases cover all the lines and branches of the added code.
- [ ] I ran `npm run build` and tested `dist/web3.min.js` in a browser.
- [ ] I have tested my code on the live network.
- [ ] I have checked the Deploy Preview and it looks correct.
- [ ] I have updated the `CHANGELOG.md` file in the root folder.
- [ ] I have linked Issue(s) with this PR in "Linked Issues" menu.

### CODE REVIEW

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+
+Before we start writing and deploying our contract, we need to set up our environment. For that, we need to install the following:
+
+1. Node.js - Node.js is a JavaScript runtime environment that allows you to run JavaScript on the server-side. You can download it from https://nodejs.org/en/download/  
+
+2. npm - Node Package Manager is used to publish and install packages to and from the public npm registry or a private npm registry. Here is how to install it https://docs.npmjs.com/downloading-and-installing-node-js-and-npm. (Alternatively, you can use yarn instead of npm https://classic.yarnpkg.com/lang/en/docs/getting-started/)
+
+# Getting started
+
+Here we initialize Web3 and set an RPC url inside `const Url;` , add our private key in the `const privateKey;` , then we specify the erc20 token inside the
```
Comment (avkos) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md]: I think it is better to not start constant name from Capital letter
Url => url

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+To sign a transaction we perform the `signTransaction` method on the account with the help of the web3.js lib.
+Next we use the `sendSignedTransaction` method with the `signedTx` parameters.
+``` 
+Javascript:
+
+const web3 = new Web3(new Web3.providers.WebsocketProvider(providerUrl));
+const Web3 = require('web3');
+
+const Url = 'wss://mainnet.infura.io/v3/******';
```
Comment (avkos) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md]: the same here

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+
+Construct the Transaction object:  
+To send the transaction details to the RPC provider for the transfer, we need to format the details in the tx object(transaction object). The tx object must contain the following:  
+from, to, value, gas, maxPriorityFeePerGas.
+Sign and send the transaction:  
+
+To sign a transaction we perform the `signTransaction` method on the account with the help of the web3.js lib.
+Next we use the `sendSignedTransaction` method with the `signedTx` parameters.
+``` 
+Javascript:
```
Comment (avkos) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:65]: We use typescript in doc examples. or both TypeScript and JavaScript as tabs like here https://docs.web3js.org/guides/advanced/tree_shaking

**Review by avkos — CHANGES_REQUESTED** [review]
Thank you for your contribution. I've written some comments

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md (hunk 3 above)
Comment (sarthak1dev) [reply docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:65]: I'll make this change asap

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md (hunk 1 above)
Comment (sarthak1dev) [reply docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md]: I am just a newbie, so thank YOU for correcting and helping me 😅. I will change this asap

**Comment by sarthak1dev** [comment]
hello ser @avkos, the changes have been made, kindly review them

**Review by avkos — APPROVED** [review]

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+<Tabs groupId='prog-lang' queryString>
+
+  <TabItem value='javascript' label='JavaScript'
+  	attributes={{className: 'javascript-tab'}}>
+
+``` 
+Javascript:
+
+const web3 = new Web3(new Web3.providers.WebsocketProvider(providerUrl));
+const Web3 = require('web3');
```
Comment (jdevcs) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:68]: web3 instance is created before its imported , pls add tested code. @sarthak1dev

**Review by jdevcs — CHANGES_REQUESTED** [review]

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+const url = 'wss://mainnet.infura.io/v3/******';
+const privateKey = 'YOUR_PRIVATE_KEY';
+const tokenAddress = '0x...'; 
+
+async function transferWithEIP1559() {
+  try {
+
+    const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+    web3.eth.accounts.wallet.add(account);
```
Comment (jdevcs) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:78]: you can directly pass private key in web3.eth.accounts.wallet.add(PK)

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 6]
```diff
@@ -0,0 +1,176 @@
...
+    // Set max priority fee (optional, you can adjust this)
+    const maxPriorityFeePerGas = web3.utils.toWei('2', 'gwei');
+
+    // Construct the transaction object
+    const tx = {
+      from: account.address,
+      to: tokenAddress, // For ERC-20 token 
+      value: amountToTransfer, 
+      gas: web3.utils.toHex(210000), // Gas limit, adjust as needed
+      gasPrice,
```
Comment (jdevcs) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:92]: in 0x2 tx there shouldnt be gasPrice, as its for 0x0 tx

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 7]
```diff
@@ -0,0 +1,176 @@
...
+    // Set max priority fee 
+    const maxPriorityFeePerGas = web3.utils.toWei('2', 'gwei');
+
+    // Construct the transaction object
+    const tx: Web3.Types.Transaction = {
+      from: account.address,
+      to: tokenAddress,
+      value: web3.utils.toWei(amountToTransfer, 'ether'), // Ensure correct unit for ERC-20 tokens
+      gas: web3.utils.toHex(210000),
+      gasPrice,
```
Comment (jdevcs) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:139]: same here,

File: docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+    const gasPrice = await web3.eth.gasPrice();
+
+    // Set max priority fee 
+    const maxPriorityFeePerGas = web3.utils.toWei('2', 'gwei');
+
+    // Construct the transaction object
+    const tx: Web3.Types.Transaction = {
+      from: account.address,
+      to: tokenAddress,
+      value: web3.utils.toWei(amountToTransfer, 'ether'), // Ensure correct unit for ERC-20 tokens
```
Comment (jdevcs) [inline docs/docs/guides/smart_contracts/erc-transfer_with-EIP-1559.md:137]: these examples are not of erc 20 transfers!

**Review by jdevcs — CHANGES_REQUESTED** [review]

**Comment by jdevcs** [comment]
I am closing this PR as there is no activity.

**Pull request closed without merging**